core/src/main/java/io/nixer/nixerplugin/core/stigma/StigmaConfiguration.java

```diff
@@ -79,26 +79,25 @@ public EncryptedJwtValidator buildEncryptedJwtValidator(final KeysLoader keysLoa
     }
 
     @Bean
-    public StigmaTokenPayloadValidator buildStigmaTokenPayloadValidator(final StigmaProperties stigmaProperties) {
-        final Duration tokenLifetime = !StringUtils.isEmpty(stigmaProperties.getTokenLifetime())
-                ? Duration.parse(stigmaProperties.getTokenLifetime())
-                : StigmaTokenConstants.DEFAULT_TOKEN_LIFETIME;
-
-        return new StigmaTokenPayloadValidator(
-                Instant::now,
-                tokenLifetime
-        );
+    public StigmaTokenPayloadValidator stigmaTokenPayloadValidator() {
+        return new StigmaTokenPayloadValidator();
     }
 
+    @Primary
     @Bean
-    public EncryptedStigmaTokenProvider buildEncryptedStigmaTokenProvider(final KeysLoader keysLoader) {
+    public EncryptedStigmaTokenProvider encryptedStigmaTokenProvider(KeysLoader keysLoader,
+                                                                     PlainStigmaTokenProvider plainStigmaTokenProvider) {
 
         return new EncryptedStigmaTokenProvider(
-                new PlainStigmaTokenProvider(Instant::now),
+                plainStigmaTokenProvider,
                 DirectEncrypterFactory.withKeysFrom(keysLoader)
         );
     }
 
+    @Bean
```

Comment:
Do you mean not making this a bean and injecting on the fly instead? It might be less verbose though it would introduce a different approach of injecting services. I'd rather keep a single way and inject all as beans.